File: c+1/starting/SpeedyBoi/python/Hash.py
from Node import Node

def main():
    x = Hash(hash)
    for i in range(1000000):
        x.add(i)
    for i in range(1000000):
        x.check(i)
    for i in range(1000000):
        x.remove(i)
    print("Done")


class Hash:
    def __init__(self,func,len = 1000000):
        self.func = func
        self.len = len
        self.size = 0
        self.nodes = [None for i in range(len)]

    def check(self, val):
        hack = self.func(val)
        if (hack < 0):
            hack *= -1
        hack %= self.len
        temp = self.nodes[hack]
        while (temp):
            if (temp.val is val):
                return True
            temp = temp.next
        return False

    def add(self, val):
        hack = self.func(val)
        if (hack < 0):
            hack *= -1
        hack %= self.len
        self.nodes[hack] = Node(val, self.nodes[hack])
        self.size+=1

    def remove(self, val):
        hack = self.func(val)
        if (hack < 0):
            hack *= -1
        hack %= self.len
        if (not self.nodes[hack]):
            return
        if (self.nodes[hack].val is val):
            self.nodes[hack] = self.nodes[hack].next
            return
        temp = self.nodes[hack]
        while(temp.next is not None):
            if (temp.next.val is val):
                temp.next = temp.next.next
                return
            temp = temp.next
if __name__ == '__main__':
    main()

File: c+1/starting/SpeedyBoi/Java/Hash.java

class Hash<T>{
    private Node<T>[] nodes;
    private int size;

    public static void main(String[] args) {
        Hash<Integer> epicness = new Hash<Integer>(10000);
        int x = 1000000;
        for (int i = 0; i < x; i++) {
            epicness.add(i);
        }
        // cout << "not noice" << endl;
        for (int i = 0; i < x; i++) {
            epicness.check(i);
            // cout << "fin" << endl;
        }
        for (int i = 0,l = x/2; i < l; i++) {
            epicness.remove(i);
        }
        System.out.println("Done");
    }

    public Hash(int len){
        nodes = new Node[len];
        size = 0;
    }

    public void add(T o){
        int hash = o.hashCode();
        if (hash < 0) {
            hash *= -1;
        }
        hash %= nodes.length;
        nodes[hash] = new Node(o, nodes[hash]);
        size++;

    }
    public boolean check(T o) {
        int hash = o.hashCode();
        if (hash < 0) {
            hash *= -1;
        }
        hash %= nodes.length;
        Node<T> temp = nodes[hash];
        while(temp != null) {
            if (temp.equals(o)) {
                return true;
            }
            temp = temp.getNext();
        }
        return false;
    }
    public int getSize(){
        return size;
    }
    public void remove(T o) {
        int hash = o.hashCode();
        if (hash < 0) {
            hash *= -1;
        }
        hash %= nodes.length;
        if (nodes[hash] == null) {
            return;
        }
        if (nodes[hash].equals(o)) {
            nodes[hash] = nodes[hash].getNext();
            size --;
            return;
        }
        Node<T> temp = nodes[hash];
        while (temp.getNext() != null) {
            if (temp.getNext().equals(o)) {
                temp.setNext(temp.getNext().getNext());
                size --;
                return;
            }
            temp = temp.getNext();
        }
    }


    public Hash() {
        this(1000000);
    }


}
